be/src/exprs/vectorized/binary_function.h

```diff
@@ -172,8 +172,10 @@ class UnionNullableColumnBinaryFunction {
 
         ColumnPtr data_result = FN::template evaluate<LType, RType, ResultType>(data1, data2);
         if (v1->has_null() || v2->has_null()) {
+            // two NOT-nullable column can generate a {nullable, not-nullable, const, const-null} column
+            // because when decimal overflow checking is enabled, overflow results will yield NULL.
             NullColumnPtr null_flags = FunctionHelper::union_nullable_column(v1, v2);
-            return NullableColumn::create(std::move(data_result), std::move(null_flags));
+            return FunctionHelper::merge_column_and_null_column(std::move(data_result), std::move(null_flags));
```

Comment:
should only affect the decimal v3 type

Reply:
Yes,  only decimal v3 type is affected
merge_column_and_null_column can deal with not-nullable data_result yields by other arithmetic operations on other types, it return data_result directly.